container/docker/factory.go

```diff
@@ -197,6 +208,80 @@ func startThinPoolWatcher(dockerInfo *dockertypes.Info) (*devicemapper.ThinPoolW
 	return thinPoolWatcher, nil
 }
 
+func getKernelVersion() (string, error) {
+	unameOutput := syscall.Utsname{}
+	err := syscall.Uname(&unameOutput)
+	if err != nil {
+		return "", err
+	}
+
+	b := make([]byte, len(unameOutput.Release))
+	for i, v := range unameOutput.Release {
+		b[i] = byte(v)
+	}
+	return string(b), nil
+}
+
+func ensureThinLsKernelVersion(kernelVersion string) error {
+	// kernel 4.4.0 has the proper bug fixes to allow thin_ls to work without corrupting the thin pool
+	version_4_4_0 := semver.MustParse("4.4.0")
+	// RHEL kernel 3.10.0 release >= 366 has the proper bug fixes backported from 4.4.0 to allow
+	// thin_ls to work without corrupting the thin pool
+	version_3_10_0 := semver.MustParse("3.10.0")
+
+	matches := version_re.FindStringSubmatch(kernelVersion)
```

Comment:
Make blows up on x86_64. Maybe there's another function that doesn't?
On Thu, Aug 4, 2016 at 1:35 PM Tim St. Clair <notifications@github.com>
wrote:

> In container/docker/factory.go
> <https://github.com/google/cadvisor/pull/1411#discussion_r73567406>:
>
> > +
> > +	b := make([]byte, len(unameOutput.Release))
> > +	for i, v := range unameOutput.Release {
> > +		b[i] = byte(v)
> > +	}
> > +	return string(b), nil
> > +}
> > +
> > +func ensureThinLsKernelVersion(
>
> kernelVersion string) error {
> > +	// kernel 4.4.0 has the proper bug fixes to allow thin_ls to work without corrupting the thin pool
> > +	version_4_4_0 := semver.MustParse("4.4.0")
> > +	// RHEL kernel 3.10.0 release >= 366 has the proper bug fixes backported from 4.4.0 to allow
> > +	// thin_ls to work without corrupting the thin pool
> > +	version_3_10_0 := semver.MustParse("3.10.0")
> > +
> > +	matches := version_re.FindStringSubmatch(kernelVersion)
>
> Is this just to check that kernelVersion has sane formatting? I think
> blang/semver already does that with semver.Make
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/google/cadvisor/pull/1411/files/d1e20410961d80cf2bde3a8a68c6249cf7c06c95#r73567406>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAABYtvojP7NzLKBOb5XYLhNiCKb-MLvks5qciL6gaJpZM4Jc2Va>
> .
>